pkg/distributor/distributor.go

```diff
@@ -218,6 +238,44 @@ func shardByAllLabels(userID string, labels []client.LabelPair) (uint32, error)
 	return h.Sum32(), nil
 }
 
+// Remove the replica label from a slice of LabelPairs if it exists.
+func removeReplicaLabel(replica []byte, labels *[]client.LabelPair) {
+	for i := 0; i < len(*labels); i++ {
+		pair := (*labels)[i]
+		if pair.Name.Equal(replica) {
+			*labels = append((*labels)[:i], (*labels)[i+1:]...)
+			return
+		}
+	}
+}
+
+// Returns an error if we should not accept samples based on the HA labels found in ts. Boolean value
+// is whether we found both HA labels. We use the boolean value to know if we want to remove the
+// replica label.
```

Comment:
I don't remember is we discussed this: why are we removing the HA labels?  Should we leave them in?  It will make failover between replicas visible to the users (and cause huge amount of series churn).  OTOH it might be useful.  WDYT?

Reply:
If we find both and want to accept the current sample, we want to send the sample on to the rest of Cortex for storage, but if we kept both we would end up with a series per replica after we do any failovers between replicas. 

If we find both labels but only send the sample on for storage with the `cluster` label we would only ever have one series per cluster, regardless of the # of replicas.

Reply:
Sure, but only one series would have a value at any one time, so in aggregations people wouldn't notice the difference, right?

Reply:
Right, but that would potentially require changes to any queries they're currently using against cortex wouldn't it?

Reply:
Let me know what you want to do here.

Reply:
Yeah lets remove, we can't deal with the series chunk.